src/main/java/com/mongodb/jdbc/MongoConnection.java

```diff
@@ -137,6 +138,7 @@ private void initializeConnection(MongoConnectionProperties connectionProperties
         this.uuidRepresentation =
                 connectionProperties.getConnectionString().getUuidRepresentation();
         this.appName = buildAppName(connectionProperties);
```

Comment:
I missed this but luckily it is in Ethan's PR :) You want to add the libmongotranslate version to the app name if it is present and can be used.

Reply:
Good catch, I added it where it checks that the cluster type is supported.